tests/cypress/integration/plugins/collapse.spec.js

```diff
@@ -9,8 +9,9 @@ test('can collapse and expand element',
     `],
     ({ get }, reload) => {
         get('h1').should(haveComputedStyle('height', '0px'))
+        get('h1').should(haveAttribute('style', 'height: 0px; overflow: hidden;'))
         get('button').click()
-        get('h1').should(haveAttribute('style', 'overflow: hidden; height: auto;'))
+        get('h1').should(haveAttribute('style', 'height: auto;'))
```

Comment:
I don't think it's correct. `overflow: hidden;` cannot be removed until the animation is complete

Reply:
right, so that's the final state, gotcha. I think cypress is too slow at checking the state since the whole animation happens very quickly.

FYI, Alpine.transition allows a callback to run before the animation start and one to run after the animation.
It will probably be the same for cypress though but it may make more sense to set and unset the overflow property in there (el.style.overflow = 'hidden' and el.style.removeProperty('overflow'), you don't need to use Alpine.setStyle if it's a single property)
```js
Alpine.transition(el, Alpine.setStyles, {
   ...
}, () => console.log('before'), () => console.log('after'))
```
This is not official feedback, of course.

Reply:
Yeah, testing things like this with Cypress is a nightmare, not the end of the world if we can't. Ideally we have a manual test in the manual_transitions_tests.html file (I think that's what it's called)

I need to have a deeper look at this before I merge though. Will look sometime this week.

Reply:
@calebporzio had any time yet? :)